src/AutoBuild.jl

```diff
@@ -71,85 +72,158 @@ function build_tarballs(ARGS, src_name, src_version, sources, script,
-        end
-        p = parse.(Int, split(ARGS[i][8:end], '/'))
-        (length(p) == 2 && p[2] > 0 && 1 ≤ p[1] ≤ p[2]) || error("invalid argument ", ARGS[i])
-        n = (length(platforms) + p[2]-1) ÷ p[2]
-        platforms = platforms[n*(p[1]-1)+1:min(end,n*p[1])]
-        deleteat!(ARGS, i)
+    # This sets whether we are going to deploy our binaries to GitHub releases
+    deploy, deploy_repo = extract_flag("--deploy", "JuliaBinaryWrappers/$(src_name)_jll.jl")
+
+    # This sets whether we are going to register, and if so, which 
+    register, register_path = extract_flag("--register", Pkg.depots1())
+    if register && !deploy
+        error("Cannot register without deploying!")
     end
+    if deploy
+        code_dir = joinpath(Pkg.depots1(), "dev", "$(src_name)_jll")
 
+        # Shove them into `kwargs` so that we are conditionally passing them along
+        kwargs = (; kwargs..., code_dir = code_dir)
+    end
+    
     # If the user passed in a platform (or a few, comma-separated) on the
     # command-line, use that instead of our default platforms
     if length(ARGS) > 0
-        should_override_platforms = true
         platforms = platform_key_abi.(split(ARGS[1], ","))
     end
 
-    # If we're running on CI (Travis, GitLab CI, etc...) and this is a
-    # tagged release, automatically determine bin_path by building up a URL
-    repo = get_repo_name()
-    tag = get_tag_name()
-
-    product_hashes = if !only_buildjl
-        # If the user didn't just ask for a `build.jl`, go ahead and actually build
-        @info("Building for $(join(triplet.(platforms), ", "))")
-
-        # Build the given platforms using the given sources
-        autobuild(pwd(), src_name, src_version, sources, script, platforms,
-                         products, dependencies; verbose=verbose, debug=debug, kwargs...)
-    else
-        msg = strip("""
-        Reconstructing product hashes from GitHub Release $(repo)/$(tag)
-        """)
-        @info(msg)
+    # Check to make sure we have the necessary environment stuff
+    if deploy
+        if !haskey(ENV, "GITHUB_TOKEN")
+            error("Must define a GITHUB_TOKEN environment variable to upload with `ghr`!")
+        end
+        if !haskey(ENV, "GITHUB_USER")
+            error("Must define a GITHUB_USER environment variable to push jll packages!")
+        end
 
-        # Reconstruct product_hashes from github
-        product_hashes_from_github_release(repo, tag;
-            product_filter="v$(src_version)",
-            verbose=verbose
-        )
+        # Check to see if we've already got a wrapper package within the Registry,
+        # choose a version number that is greater than anything else existent.
+        build_version = get_next_wrapper_version(src_name, src_version)
+        if verbose
+            @info("Building and deploying version $(build_version)")
+        end
+        repo_name = "JuliaBinaryWrappers/$(src_name)_jll.jl"
```

Comment:
This should probably be changed or removed since AFAIU this is what `deploy_repo` above is meant to be.

Reply:
Oh you are very correct